packages/web-components/src/divider/divider.template.ts

```diff
@@ -0,0 +1,8 @@
+import { ElementViewTemplate, html } from '@microsoft/fast-element';
+// import { FASTDivider } from '@microsoft/fast-foundation';
+import type { Divider } from './divider.js';
+
+/**
+ * @internal
+ */
+export const template: ElementViewTemplate<Divider> = html<Divider>`<slot></slot>`;
```

Comment:
Were this the only template necessary I think we could likely just reuse the FAST Foundation template. With that said, there is some logic in the react implementation that I don't think can be accomplished with the current template and I'm not sure we can align on it. I think we'll need to translate the react template into FAST implementation, but some of the conditional rendering comes with cost that seems a bit unnecessary relative to the benefit. Can you take a stab at converting the react template? I'd be happy to share what I did as an exploration several months ago, but I'm not super happy with the 1:1 given perf considerations. (Using a when seems like overkill...)

Reply:
You lost me.

Reply:
So our default approach is to leverage FAST Foundation where it makes sense and we are able - if there isn't alignment, can we get aligned, etc...

The Foundation template is here: https://github.com/microsoft/fast/blob/master/packages/web-components/fast-foundation/src/divider/divider.template.ts

As you can see, we need to bind the _role_ and the orientation to the template to ensure that the correct a11y roles and information gets provided. In the case of the Fluent divider for React v9 they add some structure and a slot, so we _can't_ use this outright and we need to create a template that meets the requirements for Fluent Divider. 

Here was a stab I took previously, but I think we can improve it: https://github.com/chrisdholt/fluent-v9-example-wc/blob/main/src/divider/divider.template.ts

We'll also want this to be a function like the badge...

Reply:
Ryan and I are going to work through this, then I'll commit my changes.